### DISCUSSION TITLE
Receive JSON object and reaction data

### DISCUSSION THREAD

**Discussion by nschwan94** [AsamK/signal-cli#1479] [General]
I'm recently experimenting with this software and notice that a receive JSON object like below does not contain any data on the message that is being reacted to, only the number of the reactee and the number of the reactor. I've redacted some values out but the keys are all there that I'm seeing. I wonder if I can get information on what message is being reacted to. Going to look into more.

```
{
    _id: ObjectId('65dc1a74275ac1ba374aff7b'),
    envelope: {
      source,
      sourceNumber,
      sourceUuid,
      sourceName: 'Nathan Schwan',
      sourceDevice: 1,
      timestamp: 1708923499032,
      dataMessage: {
        timestamp: 1708923499032,
        message: null,
        expiresInSeconds: 0,
        viewOnce: false,
        reaction: {
          emoji: '😂',
          targetAuthor,
          targetAuthorNumber,
          targetAuthorUuid,
          targetSentTimestamp: 1708923449303,
          isRemove: false
        }
      }
    },
    account,
  },
```

**Comment by AsamK** [comment]
Signal uses the message timestamp as id. So you can use targetSentTimestamp and targetAuthorUuid as identifier for the message that is being reacted to.

**Reply by nschwan94** [reply]
Thanks for solving.

Lmk if I can help with anything. Just a dev with some Android and Node under my belt.
